planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/persistence/GoalRepository.java

```diff
@@ -1,10 +1,21 @@
 package pl.edu.agh.kuce.planner.goal.persistence;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 import pl.edu.agh.kuce.planner.auth.persistence.User;
 
 import java.util.List;
+import java.util.Optional;
 
-public interface GoalRepository extends  JpaRepository<Goal, Integer> {
+public interface GoalRepository extends JpaRepository<Goal, Integer> {
+    @Query("SELECT g FROM Goal g where g.id = ?1")
+    Optional<Goal> getGoalById(Integer id);
+
+    @Query("SELECT g FROM Goal g where g.user = ?1")
     List<Goal> findByUser(User user);
+
+    @Modifying
+    @Query("DELETE FROM Goal g where g.id = ?1")
+    void deleteGoal(Integer id);
```

Comment:
This operation is unsafe. Client code can delete goal without authorization. Filtering also by user would fix it.